bibliopixel/drivers/spi_driver_base.py

```diff
@@ -1,75 +1,158 @@
-import os, sys, time
+import fcntl
+import math
+import os
+import struct
 from . channel_order import ChannelOrder
 from . driver_base import DriverBase
 from .. import log
 
 
-class DriverSPIBase(DriverBase):
-    """
-    Base driver for controling SPI devices on systems like the Raspberry Pi and
-    BeagleBone.
-    """
-    def __init__(self, num, c_order=ChannelOrder.GRB, use_py_spi=True,
-                 dev='/dev/spidev0.0', SPISpeed=2, open=open, gamma=None):
-        super().__init__(num, c_order=c_order, gamma=gamma)
+class SpiBaseInterface(object):
+    """ abstract class for different spi backends"""
 
-        self.dev = dev
-        self.use_py_spi = use_py_spi
-        self._spiSpeed = SPISpeed
-        self._open = open
+    def __init__(self, dev, SPISpeed):
+        self._dev = dev
+        self._spi_speed = SPISpeed
 
-        # File based SPI requires a bytearray so we have to overwrite _buf
-        if self.use_py_spi:
-            a, b = -1, -1
-            d = self.dev.replace('/dev/spidev', '')
-            s = d.split('.')
-            if len(s) == 2:
-                a = int(s[0])
-                b = int(s[1])
+    def send_packet(self, data):
+        raise NotImplementedError
 
-            if a < 0 or b < 0:
-                error(BAD_FORMAT_ERROR)
+    def compute_packet(self, data):
+        return data
 
-            self._bootstrapSPIDev()
-            import spidev
-            self.spi = spidev.SpiDev()
-            self.spi.open(a, b)
-            self.spi.max_speed_hz = int(self._spiSpeed * 1000000.0)
-            log.info('py-spidev speed @ %.1f MHz',
-                     (float(self.spi.max_speed_hz) / 1000000.0))
-        else:
-            self._buf = bytearray(self._buf)
-            self.spi = self._open(self.dev, 'wb')
-
-    def _bootstrapSPIDev(self):
-        if self.use_py_spi:
-            try:
-                import spidev
-            except:
-                error(CANT_IMPORT_SPIDEV_ERROR)
-
-        if not os.path.exists(self.dev):
+
+# partial source @see https://armbedded.taskit.de/node/318
+# @see <linux/spi/spidev.h>
+# hex numbers from good old strace(1)
```

Comment:
In the <linux/spi/spidev.h> are only macros to generate this hex numbers.
To step over this I took a look to the syscalls made on an example spi c program to have not figure out how they are generated.